orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -24,8 +22,9 @@ class ORTModule(torch.nn.Module):
     """
 
     def __init__(self, module):
-        assert isinstance(
-            module, torch.nn.Module), "'module' must be a torch.nn.Module"
+        self._validate_module_type(module)
+
+        self._module_manager = self._create_module_manager(module)
```

Comment:
```suggestion
        self._torch_module = TorchModuleFactory(module)
```
using a factory seems more familiar pattern for contributors as it is a known design pattern

Reply:
```_create_module_manager``` is a factory. It just is a function that acts as a factory as opposed to a class.

Reply:
Not being named as factory and not being a class is almost a design pattern, but not quite :) Using patterns pays off in the long run as new engineers will recognize it and make changes without help from the author.

Reply:
Since in python any variable can be dynamically assigned, having a Factory as a class (for inheritance purposes) is not necessary in my opinion. In any case, I will rename it to ```TorchModuleFactory```, but at this point I feel it is just a naming convention.